packages/xo-server/src/api/xostor.mjs

```diff
@@ -224,19 +249,34 @@ create.params = {
 
 // Also called by sr.destroy if sr.SR_type === 'linstor'
 export async function destroy({ sr }) {
-  if (sr.SR_type !== 'linstor') {
-    throw new Error('Not a XOSTOR storage')
-  }
-  const xapi = this.getXapi(sr)
-  const hosts = Object.values(xapi.objects.indexes.type.host).map(host => this.getObject(host.uuid, 'host'))
-
-  await xapi.destroySr(sr._xapiId)
-  const license = (await this.getLicenses({ productType: 'xostor' })).find(license => license.boundObjectId === sr.uuid)
-  await this.unbindLicense({
-    boundObjectId: license.boundObjectId,
-    productId: license.productId,
+  const task = this.tasks.create({
+    name: `deletion of XOSTOR: ${sr.name_label}`,
+    objectId: sr.uuid,
+    type: 'xo:xostor:destroy',
+  })
+  return task.run(async () => {
+    if (sr.SR_type !== 'linstor') {
+      throw new Error('Not a XOSTOR storage')
+    }
+    const xapi = this.getXapi(sr)
+    const hosts = Object.values(xapi.objects.indexes.type.host).map(host => this.getObject(host.uuid, 'host'))
+
+    await Task.run({ properties: { name: 'deletion of the storage', objectId: sr.uuid } }, () =>
+      xapi.destroySr(sr._xapiId)
+    )
+    const license = (await this.getLicenses({ productType: 'xostor' })).find(
+      license => license.boundObjectId === sr.uuid
+    )
+    await Task.run({ properties: { name: 'unbind the attached license' } }, () =>
+      this.unbindLicense({
+        boundObjectId: license.boundObjectId,
+        productId: license.productId,
+      })
+    )
+    return Task.run({ properties: { name: `destroy volume group on ${hosts.length} hosts` } }, () =>
```

Comment:
Prefer using `await` unless you really want to forward the result, to avoid surprising change in the future (here the task's result).